src/hostname/hostnamed.c

```diff
@@ -377,6 +380,35 @@ static int get_hardware_sku(Context *c, char **ret) {
         return strdup_to(ret, sku);
 }
 
+static int get_hardware_version(Context *c, char **ret) {
+        _cleanup_free_ char *model = NULL, *sku = NULL, *version = NULL;
+        int r;
+
+        r = get_dmi_property(c, "ID_HWVERSION", &version);
+        if (r < 0)
+                return r;
+
+        /* Suppress reporting the version field, if it's the same string as the
+         * model field, which it appears to be on various systems */
+        r = get_hardware_model(c, &model);
+        if (r < 0) {
+                if (r != -ENOENT)
+                        return r;
+        } else if (streq_ptr(version, model))
+                return -ENOENT;
+
+        /* Suppress reporting the version field, if it's the same string as the
+         * SKU field, which it appears to be on various systems */
+        r = get_dmi_property(c, "ID_SKU", &sku);
```

Comment:
Why not to use `get_hardware_sky()` here? Of course, it does slightly more than `get_dmi_property()` and it is redundant here. But, I think using `get_hardware_sku()` is better for consistency with `_model` in the above.
Note, `sd_device_get_property_value()` caches the result, hence the redundancy is not impact performance a lot.

Reply:
I changed it.